### DISCUSSION TITLE
Question about SmartContractWallet address persistency

### DISCUSSION THREAD

**Discussion by AnthonyAkentiev** [alchemyplatform/rundler#617] [Q&A] [answered]
Hello dear Alchemy. I have found discussions only here. Still I hope this is somewhat relevant.
I am using Alchemy's Account Abstraction. 

0. I know User's EOA address 
1. I calculate user's SCW address by calling AccountFactory's `getAddress()` method
2. User does no AA operations here, so his SCW is **not yet deployed** 
3. I send some tokens to that SCW address
4. User creates some AA operation 
---- later ----
5. His SCW gets deployed only here  

I have questions:
1) Is it possible that someone else can deploy SCW from other EOA address and control User's tokens? It shouldn't be possible, by I just want to make sure :-)
2) Is it possible that after some time EOA -> SCW can be changed?
So instead of controlling SCW user will control some SCW' that is different? And he will loose his token

Thank you very much for attention!

**Comment by dancoombs** [comment, marked as answer]
Hey!

For both of these questions: this is enforced by the factory that you choose for your SCW. Your factory needs to ensure that only the EOA owner could deploy a contract at a particular address. Its also responsible for ensuring that the address doesn't change between when you determine the counterfactual address and deploy the account, given that you provide the same `initCode`.

This is typically done using `Create2`.

Some examples: 
 - [`SimpleAccount`](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/samples/SimpleAccountFactory.sol) 
 - [`LightAccount`](https://github.com/alchemyplatform/light-account/blob/main/src/LightAccountFactory.sol)
